src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1861,9 +1863,12 @@ - (void)_openCommissioningWindowWithSetupPasscode:(nullable NSNumber *)setupPass
                                             queue:(dispatch_queue_t)queue
                                        completion:(MTRDeviceOpenCommissioningWindowHandler)completion
 {
+    MATTER_LOG_METRIC_BEGIN(kMetricOpenPairingWindow);
+
     if (self.isPASEDevice) {
         MTR_LOG_ERROR("Can't open a commissioning window over PASE");
         dispatch_async(queue, ^{
+            MATTER_LOG_METRIC_END(kMetricOpenPairingWindow, CHIP_ERROR_INCORRECT_STATE);
             completion(nil, [MTRError errorForCHIPErrorCode:CHIP_ERROR_INCORRECT_STATE]);
```

Comment:
Would have been less error, prone, I think, to define a block that takes a payload and a CHIP_ERROR and does the MATTER_LOG_METRIC_END and then the completion call.

Then call that block in all the places in this function where we call completion, instead of calling completion...